include/net/nrf_provisioning.h

```diff
@@ -49,23 +49,38 @@ struct nrf_provisioning_mm_change {
 };
 
 /**
- * @typedef nrf_provisioning_dmode_cb_t
- * @brief Once provisioning is done this callback is to be called.
+ * @brief nrf_provisioning callback events
  *
- * @return <0 on error,  previous mode on success.
+ * nrf_provisioning events are passed back to the nrf_provisioning_event_cb_t callback function.
  */
-typedef void (*nrf_provisioning_dmode_cb_t)(void *user_data);
+enum nrf_provisioning_event {
+	/** Provisioning process started. Client will connect to the provisioning service. */
+	NRF_PROVISIONING_EVENT_START,
+	/** Provisioning process stopped. All provisioning commands (if any) executed. */
+	NRF_PROVISIONING_EVENT_STOP,
+	/** Provisioning complete. "Finished" command received from the provisioning service. */
+	NRF_PROVISIONING_EVENT_DONE
+};
+
+/**
+ * @typedef nrf_provisioning_event_cb_t
+ * @brief Called when provisioning state changes.
+ *
+ * @param event nrf_provisioning event code.
+ * @param user_data Application specific data.
```

Comment:
```suggestion
 * @param user_data Application-specific data.
```

Reply:
@peknis Thanks for the review. I have addressed the findings.